lib/gossip/damper.js

```diff
@@ -0,0 +1,542 @@
+    return !!this.flappers[flapper.address];
+};
+
+Damper.prototype.hasStarted = function hasStarted() {
+    return this.getStatus() === Damper.Status.Started;
+};
+
+Damper.prototype.initiateSubprotocol = function initiateSubprotocol(callback) {
+    var self = this;
+    var config = this.ringpop.config;
+
+    if (!this.ringpop.dissemination.isEmpty()) {
+        this.logger.warn('ringpop damper skipped subprotocol; there are changes left to disseminate', {
+            local: this.ringpop.whoami()
+        });
+        this.emit('event', new events.DampingPrematureEvent());
+        callback();
+        return;
+    }
+
+    // TODO Stop damp timer when cluster has reached damped limits
+    if (!this._validateDampedClusterLimits()) {
+        callback();
+        return;
+    }
+
+    var flapperAddrs = this._getFlapperAddrs();
+    this.logger.info('ringpop damper started', {
+        local: this.ringpop.whoami(),
+        flappers: flapperAddrs
+    });
+    this.emit('event', new events.DamperStartedEvent());
+
+    // Select members to receive the damp-req.
+    var nVal = config.get('dampReqNVal');
+    var dampReqMembers = this.ringpop.membership.getRandomPingableMembers(
+        nVal, flapperAddrs);
+    var dampReqMemberAddrs = dampReqMembers.map(function map(member) {
+        return member.address;
+    });
+
+    // Make sure rVal is satisfiable to begin with.
+    var rVal = Math.min(config.get('dampReqRVal'), dampReqMembers.length);
+    if (rVal === 0) {
+        this.logger.warn('ringpop damper skipped subprotocol; there are not enough selectable damp req members', {
+            local: this.ringpop.whoami(),
+            flappers: flapperAddrs,
+            rVal: rVal,
+            nVal: nVal,
+            numDampReqMembers: dampReqMembers.length
+        });
+        this.emit('event', new events.DampReqUnsatisfiedEvent());
+        callback();
+        return;
+    }
+
+    this._fanoutDampReqs(flapperAddrs, dampReqMembers, rVal, onDampReqs);
+
+    function onDampReqs(err, res) {
+        if (err) {
+            self.ringpop.stat('increment', 'damper.damp-req.error');
+            self.logger.warn('ringpop damper errored out', {
+                local: self.ringpop.whoami(),
+                dampReqMembers: dampReqMemberAddrs,
+                errors: err
+            });
+            self.emit('event', new events.DampReqFailedEvent(err));
+            callback();
+            return;
+        }
+
+        self.logger.debug('ringpop damper received all damp-req responses', {
+            local: self.ringpop.whoami(),
+            flappers: flapperAddrs,
+            responses: JSON.stringify(res)
+        });
+        var membersToDamp = Damper.getMembersToDamp(res, rVal, config);
+        if (membersToDamp.length === 0) {
+            self.ringpop.stat('increment', 'damper.damp-req.inconclusive');
+            self.logger.warn('ringpop damper inconclusive', {
```

Comment:
Hm. Interesting question. Not necessarily, because if any node identifies a 'flapper' that means something "weird" is going on and the app isn't _completely_ healthy. But I understand what you mean about logging this at the warn level. `info` might be more appropriate.